Task BJS2-70506

FROLOV-KI opened the pull request:
1) Notification manager added
2) Moderator class added
3) Corrector class added

FROLOV-KI committed:
Task BJS2-70506

1) Notification manager added
2) Moderator class added
3) Corrector class added

FROLOV-KI committed:
Code style updated

iBerbasov commented on src/main/java/school/faang/lambdas/NotificationManager.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public NotificationManager registerHandler(NotificationType type, Consumer<Notification> handler) {
+        notificationHandlers.put(type, handler);
```
При обращении к полям класса хорошей практикой считается использование this
`this.notificationHandlers...`
Аналогично для остальных обращений к полям

iBerbasov commented on src/main/java/school/faang/lambdas/moderator/DumbModerator.java:
```diff
@@ -0,0 +1,14 @@
+
+    public DumbModerator() {
+        rules = List.of(
+                new Rule("Rule 1", "Usage of bad word 1 forbidden",
```
Каждый аргумент лучше писать на новой строке, так более очевидны все аргументы конструктора
```
new Rule("Rule 1", 
    "Usage of bad word 1 forbidden",
     notification -> notification.message().contains("bad word 1"))
```

iBerbasov commented on src/main/java/school/faang/lambdas/NotificationManager.java:
```diff
@@ -0,0 +1,46 @@
+    public final Corrector corrector;
+    private final Map<NotificationType, Consumer<Notification>> notificationHandlers = new HashMap<>();
+
+    public NotificationManager(Moderator moderator) {
```
Этот конструктор нигде не используется. Если он не нужен, то лучше удалить

iBerbasov commented on src/main/java/school/faang/lambdas/Main.java:
```diff
@@ -0,0 +1,56 @@
+        } catch (IllegalMessageException e) {
+            log.warn(e.getMessage());
+        }
+
```
лишняя пустая строка

iBerbasov commented on src/main/java/school/faang/lambdas/Main.java:
```diff
@@ -0,0 +1,56 @@
+        try {
+            amazonMessageManager.sendNotification(
+                    new Notification(NotificationType.EMAIL, "send email with bad word 2 to you"));
+
```
Лишняя пустая строка

iBerbasov commented on src/main/java/school/faang/lambdas/NotificationManager.java:
```diff
@@ -0,0 +1,46 @@
+        this.corrector = corrector;
+    }
+
+    public NotificationManager registerHandler(NotificationType type, Consumer<Notification> handler) {
```
Классное решение!)

iBerbasov commented on src/main/java/school/faang/lambdas/corrector/DefaultCorrector.java:
```diff
@@ -0,0 +1,8 @@
+package school.faang.lambdas.corrector;
+
+public class DefaultCorrector extends Corrector {
```
Всё же название DefaultCorrector предусматривает, что будут производится какие-то базовые действия. Этот дефолтный корректор ничего не делает.
Рассмотри варианты переименовать его, чтобы имя подсказывало, что корректировки никакой не будет. мне фантазия подсказала `NoCorrectionCorrector`, может у тебя появятся варианты лучше?:)
Или вообще убрать его, а в основной логике предусмотреть отсутствие корректора

iBerbasov reviewed (commented):
Решение крутое! Исправить мелкие недочёты и будет конфетка!

FROLOV-KI committed:
PR issues resolved

1) Default corrector updated
2) NoCheckModerator added
3) Notification manager constructor updated
4) Moderator constructor updated

FROLOV-KI committed:
Code style fixed

FROLOV-KI committed:
minor fix
